test/jdk/javax/swing/JList/bug4193267.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2001, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4193267
+ * @summary Tests that JList first and last visible indices are
+ * updated properly
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug4193267
+ */
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextField;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.GridLayout;
+
+public class bug4193267 {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+            Resize the frame "JList" with a different ways and scroll the list
+            (if it possible). The indices of first and last visible elements
+            should be indicated in the corresponding fields in "Index" frame.
+            If the indicated indices is not right then test fails.
+
+            Note:
+              - the first and last visible indices should be -1 if nothing
+              is visible;
+              - the first or last visible cells may only be partially visible.
+            """;
+        PassFailJFrame.builder()
+            .title("bug4193267 Instructions")
+            .instructions(INSTRUCTIONS)
+            .columns(35)
+            .testUI(initialize())
```

Comment:
PassFailJFrame's  ` .positionTestUI(WindowLayouts::rightOneRow)` can be used to position multiple windows side by side.

Reply:
Updated.